site/content/docs/user-guide/setup/install.md

```diff
@@ -22,9 +22,11 @@ Krew self-hosts).
     ```sh
     (
       set -x; cd "$(mktemp -d)" &&
+      OS="$(uname | tr '[:upper:]' '[:lower:]')" &&
+      ARCH=`case "${OS}" in "darwin") uname -m | sed -e 's/x86_64/amd64/' ;; *) uname -m | sed -e 's/x86_64/amd64/' -e 's/arm.*$/arm/' -e 's/aarch64$/arm64/' ;; esac` &&
```

Comment:
this part is a little suspicious: why did you need `case "${OS}" in "darwin")` ?

it seems that `uname -m` already returns arm64 in Apple Silicon chip, so it would work as is?

if you did this because of `-e 's/arm.*$/arm/'`, that's understandable as it could return values like `armv7l`, `armv8` in other platforms such as Linux. in that case, we can rewrite the regexp pattern in a way that does not replace `amd64` (coming from M1).

Reply:
Yeah, this is the part that I mentioned in the PR description.

> The install scripts are a bit more complicated now because uname -m on Apple Silicon returns arm64, so the sed -e 's/arm.*$/arm/' substitution resulted in darwin_arm, rather than darwin_arm64.
>
> I fixed that as best I could with a case statement, but perhaps there's a simpler approach. I looked into a pure sed solution, but it doesn't seem like sed supports the necessary lookahead features that would enable search/replace for arm + anything other than 64.

Any better ideas?

Reply:
Pushed a new update with a better sed-only replacement that seems to work. WDYT?